usecases/schema/tenant.go

```diff
@@ -255,15 +255,17 @@ func (m *Manager) GetTenants(ctx context.Context, principal *models.Principal, c
 		return nil, fmt.Errorf("multi-tenancy is not enabled for class %q", class)
 	}
 
-	m.shardingStateLock.Lock()
 	var tenants []*models.Tenant
-	if ss := m.state.ShardingState[cls.Class]; ss != nil {
-		tenants = make([]*models.Tenant, 0, len(ss.Physical))
+	m.schemaCache.RLock()
+	if ss := m.schemaCache.ShardingState[cls.Class]; ss != nil {
+		tenants = make([]*models.Tenant, len(ss.Physical))
+		i := 0
 		for tenant := range ss.Physical {
-			tenants = append(tenants, &models.Tenant{Name: tenant})
+			tenants[i] = &models.Tenant{Name: tenant}
+			i++
 		}
 	}
-	m.shardingStateLock.Unlock()
+	m.schemaCache.RUnlock()
```

Comment:
Can we also here make use of `m.schemaCache.RockGuard()` method? instead of manual locking?

Reply:
Agreed